libres/tests/analysis/test_compare_initX.cpp

```diff
@@ -0,0 +1,120 @@
+#include <catch2/catch.hpp>
+#include <vector>
+#include <cmath>
+
+#include <ert/util/rng.h>
+#include <ert/enkf/enkf_util.hpp>
+
+#include <ert/util/bool_vector.hpp>
+#include <ert/enkf/obs_data.hpp>
+#include <ert/enkf/meas_data.hpp>
+#include <ert/analysis/ies/ies_enkf.hpp>
+#include <ert/analysis/std_enkf.hpp>
+
+struct model {
+    double amplitude;
+    double phase;
+
+    model(double a, double p) : amplitude(a), phase(p) {}
+
+    model(rng_type *rng) {
+        this->amplitude = 0.25 + enkf_util_rand_normal(1, 0.5, rng);
+        this->phase = enkf_util_rand_normal(0, M_PI / 4, rng);
+    }
+
+    double eval(double x) const {
+        return this->amplitude * std::sin(x + this->phase);
+    }
+};
+
+TEST_CASE("compare_initX", "[analysis]") {
+    const char *obs_key = "OBS1";
+    const int ens_size = 100;
+    const int obs_size = 10;
+    std::vector<model> ens;
+    bool_vector_type *ens_mask = bool_vector_alloc(ens_size, true);
+    obs_data_type *obs_data = obs_data_alloc(1.0);
+    meas_data_type *meas_data = meas_data_alloc(ens_mask);
+    rng_type *rng = rng_alloc(MZRAN, INIT_DEFAULT);
+    model true_model{1.0, 0.0};
+
+    for (int iens = 0; iens < ens_size; iens++)
+        ens.emplace_back(rng);
+
+    meas_block_type *mb = meas_data_add_block(meas_data, obs_key, 1, obs_size);
+    obs_block_type *ob =
+        obs_data_add_block(obs_data, obs_key, obs_size, nullptr, false);
+
+    for (int iobs = 0; iobs < obs_size; iobs++) {
+        double x = iobs * 2 * M_PI / (obs_size - 1);
+        for (int iens = 0; iens < ens_size; iens++) {
+            const auto &m = ens[iens];
+            meas_block_iset(mb, iens, iobs, m.eval(x));
+        }
+
+        obs_block_iset(ob, iobs, true_model.eval(x), 0.25);
+    }
+
+    GIVEN("Simulation results") {
+        matrix_type *S = meas_data_allocS(meas_data);
+        matrix_type *R = obs_data_allocR(obs_data);
+        matrix_type *E = obs_data_allocE(obs_data, rng, ens_size);
+        matrix_type *D = obs_data_allocD(obs_data, E, S);
+
+        WHEN("different truncations are used") {
+            std_enkf_data_type *std_enkf_data =
+                static_cast<std_enkf_data_type *>(std_enkf_data_alloc());
+
+            matrix_type *X1 = matrix_alloc(ens_size, ens_size);
+            matrix_type *X2 = matrix_alloc(ens_size, ens_size);
+            std_enkf_set_double(std_enkf_data, ENKF_TRUNCATION_KEY_,
```

Comment:
Any reason not to use `std_enkf_set_truncation` directly, instead of the wrapper `std_enkf_set_double`?

Also, I tried running the test with truncation set to `1.0` and to `0.01`:

`std_enkf_set_truncation(std_enkf_data, GENERATE(1.0, DEFAULT_ENKF_TRUNCATION_, 0.90, 0.01));`

The tests pass.
Does that seem reasonable to you?

Reply:
> Any reason not to use std_enkf_set_truncation directly, instead of the wrapper std_enkf_set_double

Historical reason - closer to user-config facing implementation of `analysis_module_set_var()`. Have changed it.

Reply:
>The tests pass.
Does that seem reasonable to you?

I guess you are alluding to the values 1.0 and 0.01? The value of 1.0 will mean that all singular values in the SVD are used - this will potentially become quite numerical unstable and prone to overfitting. The value of 0.01 means that only 1% of the variation will be included - essentially that will imply no singular values, and not very meaningful results. But - the current test just verifies that two different methods to initialize the X matrix give the same result, it does not in any way evaluate the quality of the results. So - yes I would expect this test to pass even if somewhat weird truncation values are used.

Reply:
I was alluding to 1.0 and 0.01.
Perhaps we don't need to test with different truncations if they don't matter?

Reply:
> Perhaps we don't need to test with different truncations if they don't matter?

Well - there is a point in testing with different truncations to verify that the truncation value can be modified for both `initX()` implementations.